CHANGELOG.rst

```diff
@@ -1,3 +1,16 @@
+2.1.0 (2024-02-21)
+------------------
+
+* `#243 <https://github.com/pytest-dev/execnet/pull/243>`__: Added main_thread_only execmodel
+  which is derived from the thread execmodel and only executes ``remote_exec`` calls in the
+  main thread. Callers of ``remote_exec`` must use the returned channel to wait for a task to
+  complete before they call remote_exec again, otherwise the ``remote_exec`` call will fail
+  with a ``concurrent remote_exec would cause deadlock`` error. The main_thread_only execmodel
+  provides solutions for `#96 <https://github.com/pytest-dev/execnet/issues/96>`__ and
+  `pytest-dev/pytest-xdist#620 <https://github.com/pytest-dev/pytest-xdist/issues/620>`__.
```

Comment:
We need to make a new pytest-xdist release which uses `main_thread_only` correct?

```suggestion
  `pytest-dev/pytest-xdist#620 <https://github.com/pytest-dev/pytest-xdist/issues/620>`__ (pending a new `pytest-xdist` release).
```

Reply:
Correct, I have a pytest-xdist patch available in https://github.com/zmedico/pytest-xdist/pull/1.

Reply:
Applied, thanks!